components/dashboard/src/workspaces/Workspaces.tsx

```diff
@@ -28,46 +39,121 @@ export class Workspaces extends React.Component<WorkspacesProps, WorkspacesState
                     <div className="py-4">
                         <svg xmlns="http://www.w3.org/2000/svg" width="14" height="14" viewBox="0 0 490 490">
-                            <path fill="none" stroke="#777" stroke-width="36" stroke-linecap="round"
+                            <path fill="none" stroke="#999" strokeWidth="36" strokeLinecap="round"
                                 d="m280,278a153,153 0 1,0-2,2l170,170m-91-117 110,110-26,26-110-110" />
                         </svg>
                     </div>
-                    <input className="border-0" type="text" placeholder="Search Workspace" onChange={(v) => { console.log(v) }} />
+                    <input className="border-0" type="text" placeholder="Search Workspace" onChange={(v) => { if (wsModel) wsModel.setSearch(v.target.value) }} />
                 </div>
                 <div className="flex-1" />
                 <div className="py-3">
                     <DropDown entries={[{
                         title: 'Active',
                         onClick: onActive
                     }, {
-                        title: 'Past',
-                        onClick: onRecent
+                        title: 'All',
+                        onClick: onAll
+                    }]} />
+                </div>
+                <div className="py-3 pl-3">
+                    <DropDown entries={[{
+                        title: 'Max 50',
+                        onClick: () => { if (wsModel) wsModel.limit = 50; }
+                    }, {
+                        title: 'Max 100',
+                        onClick: () => { if (wsModel) wsModel.limit = 100; }
+                    }, {
+                        title: 'Max 200',
+                        onClick: () => { if (wsModel) wsModel.limit = 200; }
                     }]} />
                 </div>
             </div>
-            <div className="lg:px-28 px-10 flex flex-col space-y-2">
-                <div className="py-3 flex justify-between space-x-2 text-sm text-gray-400 border-t border-b border-gray-200">
-                    <div className="w-10"></div>
-                    <div className="w-3/12">Name</div>
-                    <div className="w-4/12">Context</div>
-                    <div className="w-2/12">Pending Changes</div>
-                    <div className="w-2/12">Last Active</div>
-                    <div className="w-8"></div>
+            {wsModel && (
+                this.state?.workspaces.length > 0 || wsModel.searchTerm ?
+                    <div className="lg:px-28 px-10 flex flex-col space-y-2">
+                        <div className="px-6 py-3 flex justify-between space-x-2 text-sm text-gray-400 border-t border-b border-gray-200">
+                            <div className="w-6"></div>
+                            <div className="w-3/12">Name</div>
+                            <div className="w-4/12">Context</div>
+                            <div className="w-2/12">Pending Changes</div>
```

Comment:
**thought**: `X Changes` below with the text-gitpod-kumquat color seems like. Maybe let's go with a red one. WDYT?